explore/src/explore.cpp

```diff
@@ -316,21 +323,61 @@ void Explore::reachedGoal(const actionlib::SimpleClientGoalState& status,
   oneshot_ = relative_nh_.createTimer(
       ros::Duration(0, 0), [this](const ros::TimerEvent&) { makePlan(); },
       true);
-  
+
   // change the state by publishing the goal reached event
-  if (currentDragoonState == EXPLORE_STATE){
+  if (currentDragoonState == EXPLORE_STATE) {
     dragoon_messages::stateCmd stateMsg;
     stateMsg.event = "GOAL REACHED";
     stateMsg.value = true;
     statePublisher_.publish(stateMsg);
   }
 }
 
+void Explore::reachedLastGoal(
+    const actionlib::SimpleClientGoalState& status,
+    const move_base_msgs::MoveBaseResultConstPtr& result,
+    const geometry_msgs::Point& frontier_goal)
+{
+  sendLastSweepAndStop();
+}
+
 void Explore::start()
 {
   exploring_timer_.start();
 }
 
+void Explore::processEndOfExplore()
+{
+  move_base_msgs::MoveBaseGoal goal;
+  goal.target_pose.pose.position = prev_goal_;
+  goal.target_pose.pose.orientation.w = 1.;
+  goal.target_pose.header.frame_id = costmap_client_.getGlobalFrameID();
+  goal.target_pose.header.stamp = ros::Time::now();
+
+  // send goal to move_base the last goal
+  move_base_client_.sendGoal(
+      goal, [this](const actionlib::SimpleClientGoalState& status,
+                   const move_base_msgs::MoveBaseResultConstPtr& result) {
+        reachedLastGoal(status, result, prev_goal_);
+      });
+
+  // Add time limit on Dragoon's attempt to go to the last goal
+  // Because the last goal might not be reachable
+  ros::Duration(lastGoalTimeLimit_).sleep();
```

Comment:
So this will block this thread for 10 seconds regardless of whether we reach the goal or not?